pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+
+        """
+        self._validate_instances_namespace(namespace)
+
+        # TODO is there any case were we will get a string classname here???
+        cname_input = params['ClassName']
+        cname = cname_input.classname if isinstance(cname_input, CIMClassName) \
+            else cname_input
+
+        # TODO implement subclasses if class exists in namespace
+        # inst_paths = [inst.path for inst in self.instances[namespace]
+        #              if inst.path.classname == cname]
+
+        inst_paths = self._enumerate_instancenames(cname, namespace)
+        rtn_paths = [copy.copy(path) for path in inst_paths]
+
+        return self._make_tuple(rtn_paths)
+
+    def _fake_execquery(self, namespace, **params):
+        """
+        Executes the equilavent of the WBEMConnection ExecQuery for
+        the querylanguage and query defined
+        """
+        raise CIMError(CIM_ERR_NOT_SUPPORTED, 'Not Implemented!')
+
+    #####################################################################
+    #
+    #  Faked WBEMConnection Reference and Associator methods
+    #
+    #####################################################################
+
+    def _get_reference_classnames(self, classname, namespace,
+                                  result_class, role):
+        """
+        Get list of classnames that are references for which this classname
+        is a target filtered by the result_class and role parameters if they
+        are none.
+        This is a common method used by all of the other reference and
+        associator methods to create a list of reference classnames
+
+        parameters:
+          ResultClass (:term:`string` or :class:`~pywbem.CIMClassName`):
+            Class name of an association class (case independent),
+            to filter the result to include only traversals of that association
+            class (or subclasses).
+
+            `None` means that no such filtering is peformed.
+
+            The ResultClass acts as a filter on the returned set of classes
+            by mandating that each returned object shall be this class
+            (or one of its subclasses).
+
+          Role (:term:`string`):
+            Role name (= property name) of the source end (case independent),
+            to filter the result to include only traversals from that source
+            role.
+
+            `None` means that no such filtering is peformed.
+
+            The Role input parameter, acts as a filter on the returnedset of
+            objects by mandating that each returned object shall refer to the
+            target object through a property with a name that matches the
+            value of this parameter.
+
+        Returns:
+            list of classnames that satisfy the criteria.
+        """
+        if not self.classes[namespace]:
+            raise CIMError(CIM_ERR_INVALID_NAMESPACE,
+                           'Namespace %s not found.' % namespace)
```

Comment:
There is a namespace validation function that could be used instead.

Reply:
I have some duplication where I recheck in internal functions. Once code stabilizes we can remove
those since the namespace validation should be at the boundaries.

Added couple of TODOs